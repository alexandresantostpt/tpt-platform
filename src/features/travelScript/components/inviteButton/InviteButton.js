import PropTypes from 'prop-types'
import React from 'react'
import { i18n } from '@i18n'
import AddButton from './AddButton'
import StyledInviteButton from './StyledInviteButton'

const InviteButton = ({ onClick, round }) => {
    if (round) {
        return <AddButton onClick={onClick}>+</AddButton>
    }

    return <StyledInviteButton onClick={onClick}>{i18n.t('buttons.invite')}</StyledInviteButton>
}

InviteButton.defaultProps = {
    round: false
}

InviteButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    round: PropTypes.bool
}

export default InviteButton
