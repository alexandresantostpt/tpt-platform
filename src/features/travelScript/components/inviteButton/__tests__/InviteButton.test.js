import React from 'react'
import renderer from 'react-test-renderer'

import InviteButton from '../InviteButton'

describe('Tests for InviteButton component', () => {
    it('Should render default button', () => {
        const tree = renderer.create(<InviteButton onClick={null} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('Should render round button', () => {
        const tree = renderer.create(<InviteButton onClick={null} round />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
