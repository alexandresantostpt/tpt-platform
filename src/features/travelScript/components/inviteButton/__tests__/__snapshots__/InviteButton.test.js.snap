// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for InviteButton component Should render default button 1`] = `
<button
  className="StyledInviteButton-sc-1iige8l-0 cJgVMw"
  onClick={null}
>
  Convidar
</button>
`;

exports[`Tests for InviteButton component Should render round button 1`] = `
<div
  className="AddButton-sc-3y6odv-0 cOrHuk"
  onClick={null}
>
  +
</div>
`;
